substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/ResourceConfigurationParser.java

```diff
@@ -47,6 +47,7 @@ public void parseAndRegister(Reader reader) throws IOException {
         parseTopLevelObject(asMap(json, "first level of document must be an object"));
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
Let's no use this. Revert to your original from where you had the cast to `Map<String, Object>`. While it is ugly its still better than suppress unchecked-warnings in the whole method body.

Reply:
The CI seems to fail now because of the unchecked cast which is considered an error. That's why I put the annotation in the first place.

Reply:
Yea. I figured. Re-added the SuppressWarnings and fixed the merge conflict.

Because I fixed the merge conflict the PR did not close automatically when the changes got merged on master.
See: https://github.com/oracle/graal/commit/c1ccd5a2436a02f3d61a22bfeb0dbf5c3bd4d8be

Thank you for your contribution!